### ISSUE TITLE
unsupported video message type when pulling from nginx-rtmp

### ISSUE THREAD

**Issue by teowoz** [bluenviron/mediamtx#2384]
## Which version are you using?

v1.1.0

## Which operating system are you using?

- [x] Linux amd64 standard
- [ ] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue

When MediaMTX tries to open rtmp:// input coming from FFmpeg through [nginx-rtmp-module](https://github.com/arut/nginx-rtmp-module), `unsupported video message type` message is displayed in log and it fails.

Type ids vary, I've encountered 15, 25, 26 and 27.

(before upgrade from 1.0.3 to 1.1.0, the error was `unsupported video codec` with id 6 or 13)

In some cases (when nginx-rtmp runs on localhost, but I'm not sure whether it's the actual reason), `stopped: timed out` is logged instead - I've already reported it as #2383

## Describe how to replicate the issue

#2383 blocks me from writing detailed replication instructions, so I'll skip nginx-rtmp stream setup.

1. start the server: `MTX_API=true MTX_APIADDRESS=:9997 MTX_LOGLEVEL=debug ./mediamtx mediamtx.yml`
2. add pull stream using the API: `curl -vv -d '{"source":"rtmp://my-nginx-rtmp-server/stream_test/bunny","sourceOnDemand":true}' -X POST -H 'Content-Type: application/json' http://localhost:9997/v2/config/paths/add/test`
3. try to read with ffprobe: `ffprobe rtmp://localhost/test`

## Did you attach the server logs?

yes, [unsupported_video_codec.log](https://github.com/bluenviron/mediamtx/files/12674279/unsupported_video_codec.log)

## Did you attach a network dump?

yes, 
[unsupported_video_codec.pcap.gz](https://github.com/bluenviron/mediamtx/files/12674312/unsupported_video_codec.pcap.gz)

**Comment by teowoz** [comment]
I managed to reproduce similar error on localhost, by modifying nginx-rtmp config. It turns out that `wait_video on;` breaks MediaMTX:
```
2023/11/15 11:57:00 DEB [path fromnginx] [RTMP source] connecting
2023/11/15 11:57:01 ERR [path fromnginx] [RTMP source] invalid body size
2023/11/15 11:57:06 DEB [path fromnginx] [RTMP source] connecting
2023/11/15 11:57:07 ERR [path fromnginx] [RTMP source] invalid body size
2023/11/15 11:57:10 INF [path fromnginx] [RTMP source] stopped: timed out
```

## How to replicate the issue
Tested with MediaMTX 1.3.0.

1. create `nginx.conf` with content:
```
events {}

rtmp {
    server {
        listen 1935;
        application live {
                live on;
                wait_video on;
        }
    }
}
```
2. start nginx-rtmp on a **non-standard port** and with our customized config, e.g. with Docker: `docker run -d -p 1936:1935 -v $PWD/nginx.conf:/etc/nginx/nginx.conf:ro --name nginx-rtmp tiangolo/nginx-rtmp`
3. transmit to nginx-rtmp: `ffmpeg -re -f lavfi -i testsrc2=r=25 -f lavfi -i anoisesrc=color=brown -c:v libx264 -g 25 -c:a aac -f flv rtmp://127.0.0.1:1936/live/aaaaaa`
4. start MediaMTX: `MTX_API=true MTX_APIADDRESS=:9997 MTX_LOGLEVEL=debug ./mediamtx mediamtx.yml` (config is the default provided with binary release package)
5. add pull stream using the API: `curl -vv -d '{"source":"rtmp://127.0.0.1:1936/live/aaaaaa","sourceOnDemand":true}' -X POST -H 'Content-Type: application/json' http://localhost:9997/v2/config/paths/add/fromnginx`
6. try to probe stream from MediaMTX: `ffprobe rtmp://localhost/fromnginx`, ffprobe will return `Input/output error` and MediaMTX will show `invalid body size`
7. see that the same stream is readable using ffprobe directly: `ffprobe rtmp://127.0.0.1:1936/live/aaaaaa`

When trying to stream from the nginx-rtmp instance I was using when reporting this bug for the first time, different error is printed in MediaMTX log: `unsupported video codec: 8`. But disabling `wait_video` in its config also fixes it.

**Comment by aler9** [comment]
Hello, this should be fixed by #2808. Please test this nightly release and let me know if it works (click on artifacts, binaries):
https://github.com/bluenviron/mediamtx/actions/runs/7207849258
